### PR TITLE
Added input set warnings and removed some unused INCAR flags

### PR DESCRIPTION
**Pull request by arosen93**
## Summary
- Copies over some of the warning messages that I recently added to `pymatgen.io.vasp.sets`
   - Note: I doubt `constrain_total_magmoms` works as expected in practice. If there is a multi-step flow, the second/third/etc. jobs will inherit the magmoms from the prior job, and then `NUPDOWN` will no longer be an integer like it's supposed to be.
- I added a function call that removes "unused" INCAR parameters: if `NSW = 0`, then remove `EDIFFG`/`IBRION`/`ISIF`/`POTIM`; if `ISPIN = 1`, then remove `MAGMOM`; if `LDAU = False`, then remove the `LDAUU`/`LDAUJ`/`LDAUTYPE` flags -- all unless the user has explicitly set them for some unusual reason.
   - While this is entirely a cosmetic thing, I do think it helps for readability's sake when investigating the INCARs. I also think it's clearer for new users. For instance, right now, if `ISPIN = 1` (e.g. when `auto_ispin = True`), the `MAGMOM` flag in the INCAR might also be set, which could be a bit counterintuitive for a new user of VASP.

~This is in draft state because I probably should write up a test or two (and perhaps get in the habit of squashing my commits).~

### CODE REVIEW

**Comment by arosen93** [comment]
All ready for your review, @utf. Note that I removed `IBRION = 2` from the `updates` in `vasp.sets.core.HSEStaticSetGenerator`. I imagine that was just a copy/paste thing.

**Review by utf — COMMENTED** [review]
One minor comment! Thanks very much.

File: src/atomate2/vasp/sets/base.py [hunk 1]
```diff
@@ -567,6 +590,15 @@ def _get_incar(
 
         if self.constrain_total_magmom:
             nupdown = sum(mag if abs(mag) > 0.6 else 0 for mag in incar["MAGMOM"])
+            if nupdown != round(nupdown):
```
[the remark this answers is not in the document]
Comment (arosen93) [reply src/atomate2/vasp/sets/base.py:593]: Sure, I'm happy to (although I think if `NUPDOWN` were indeed a value like 1.000001, that'd still be questionable to provide to VASP...).

I made the change. Just changed the equality sign from `<` to `>` compared to what you suggested.

**Comment by utf** [comment]
Brilliant. Thank you.

**Pull request merged**